crates/sui-framework-build/src/compiled_package.rs

```diff
@@ -130,35 +130,50 @@ impl CompiledPackage {
         })
     }
 
-    /// Return the bytecode modules in this package, topologically sorted in dependency order
-    /// This is the function to call if you would like to publish or statically analyze the modules
-    pub fn get_dependency_sorted_modules(&self) -> Vec<CompiledModule> {
-        let compiled_modules = self.package.root_modules_map();
-        // Collect all module IDs from the current package to be
-        // published (module names are not sufficient as we may
-        // have modules with the same names in user code and in
-        // Sui framework which would result in the latter being
-        // pulled into a set of modules to be published).
-        // For each transitive dependent module, if they are not to be published,
-        // they must have a non-zero address (meaning they are already published on-chain).
-        let self_modules: HashSet<ModuleId> = compiled_modules
-            .iter_modules()
-            .iter()
-            .map(|m| m.self_id())
-            .collect();
-        self.package
-            .all_modules_map()
-            .compute_dependency_graph()
-            .compute_topological_order()
-            .unwrap() // safe because package built successfully
-            .filter(|m| self_modules.contains(&m.self_id()))
-            .cloned()
-            .collect()
+    /// Return the bytecode modules in this package, topologically sorted in dependency order.
+    /// Optionally include dependencies that have not been published (are at address 0x0), if
+    /// `with_unpublished_deps` is true. This is the function to call if you would like to publish
+    /// or statically analyze the modules.
+    pub fn get_dependency_sorted_modules(
+        &self,
+        with_unpublished_deps: bool,
+    ) -> Vec<CompiledModule> {
+        let all_modules = self.package.all_modules_map();
+        let graph = all_modules.compute_dependency_graph();
+
+        // SAFETY: package built successfully
+        let modules = graph.compute_topological_order().unwrap();
+
+        if with_unpublished_deps {
```

Comment:
This is the heart of the change -- including all modules whose self address is `0x0`, instead of just the modules in the root package.